Code/Tools/AssetProcessor/native/utilities/Builder.cpp

```diff
@@ -0,0 +1,370 @@
+            }
+            else
+            {
+                AZ_Error("Builder", false, "AssetBuilder failed to connect within %d seconds", m_startupWaitTimeS);
+            }
+
+            return AZ::Failure(AZStd::string::format("Connection failed to builder %.*s", AZ_STRING_ARG(UuidString())));
+        }
+
+        return AZ::Success();
+    }
+
+    void Builder::SetConnection(AZ::u32 connId)
+    {
+        m_connectionId = connId;
+        m_connectionEvent.release();
+    }
+
+    AZ::u32 Builder::GetConnectionId() const
+    {
+        return m_connectionId;
+    }
+
+    AZ::Uuid Builder::GetUuid() const
+    {
+        return m_uuid;
+    }
+
+    AZStd::string Builder::UuidString() const
+    {
+        return m_uuid.ToString<AZStd::string>(false, false);
+    }
+
+    void Builder::PumpCommunicator() const
+    {
+        if (m_tracePrinter)
+        {
+            m_tracePrinter->Pump();
+        }
+    }
+
+    void Builder::FlushCommunicator() const
+    {
+        if (m_tracePrinter)
+        {
+            // flush both STDOUT and STDERR
+            m_tracePrinter->WriteCurrentString(true);
+            m_tracePrinter->WriteCurrentString(false);
+        }
+    }
+
+    void Builder::TerminateProcess(AZ::u32 exitCode) const
+    {
+        if (m_processWatcher)
+        {
+            m_processWatcher->TerminateProcess(exitCode);
+        }
+    }
+
+    AZ::Outcome<void, AZStd::string> Builder::Start(BuilderPurpose purpose)
+    {
+        // Get the current BinXXX folder based on the current running AP
+        QString applicationDir = QCoreApplication::instance()->applicationDirPath();
+
+        // Construct the Builders subfolder path
+        AZStd::string buildersFolder;
+        AzFramework::StringFunc::Path::Join(applicationDir.toUtf8().constData(), BuildersFolderName, buildersFolder);
+
+        // Construct the full exe for the builder.exe
+        const AZStd::string fullExePathString =
+            QDir(applicationDir).absoluteFilePath(AssetProcessor::s_assetBuilderRelativePath).toUtf8().constData();
+
+        if (m_quitListener.WasQuitRequested())
+        {
+            return AZ::Failure(AZStd::string("Cannot start builder, quit was requested"));
+        }
+
+        const AZStd::vector<AZStd::string> params = BuildParams("resident", buildersFolder.c_str(), UuidString(), "", "", purpose);
+
+        m_processWatcher = LaunchProcess(fullExePathString.c_str(), params);
+
+        if (!m_processWatcher)
+        {
+            return AZ::Failure(AZStd::string::format("Failed to start process watcher for Builder %.*s.", AZ_STRING_ARG(UuidString())));
+        }
+
+        m_tracePrinter = AZStd::make_unique<ProcessCommunicatorTracePrinter>(m_processWatcher->GetCommunicator(), "AssetBuilder");
+
+        return WaitForConnection();
+    }
+
+    bool Builder::IsValid() const
+    {
+        return m_connectionId != 0 && IsRunning();
+    }
+
+    bool Builder::IsRunning(AZ::u32* exitCode) const
```

Comment:
This is just a wrapper around IsProcessRunning to avoid having to do a bunch of safety checks every time.  That kind of change feels like something the ProcessWatcher API should handle/provide.